ccm_web/server/src/invitation/cirrus-invitation.service.ts

```diff
@@ -0,0 +1,76 @@
+import axios from 'axios'
+import { randomUUID } from 'crypto'
+import FormData from 'form-data'
+import { lastValueFrom } from 'rxjs'
+import { HttpService } from '@nestjs/axios'
+import { ConfigService } from '@nestjs/config'
+import { Injectable } from '@nestjs/common'
+
+import { CirrusAPIEndPoint, cirrusAPIVersion, CirrusErrorData, CirrusInvitationResponse } from './cirrus-invitation.interfaces'
+import { InvitationAPIError } from './invitation.errors'
+
+import { Config } from '../config'
+import baseLogger from '../logger'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+@Injectable()
+export class CirrusInvitationService {
+  url: string
+  entityID: string
+  sponsorName: string
+  key: string
+  secret: string
+
+  constructor (
+    private readonly configService: ConfigService<Config, true>,
+    private readonly httpService: HttpService
+  ) {
+    const invitationConfig = configService.get('invitation', { infer: true })
+    const apiURL = invitationConfig.apiURL
+    this.url = apiURL + cirrusAPIVersion + CirrusAPIEndPoint
```

Comment:
Codacy says this is an issue, but I don't think it is. I believe Codacy doesn't have access to typing from NestJS, so it can't determine the type of `apiURL`. Anywho, you can ignore this issue in Codacy. Click on the settings icon dropdown on the issue in the Codacy UI, then click "Ignore Issue".

Reply:
Thanks for the advice on ignoring.  I didn't try that, but I had tried to change the code pattern config.  That didn't work because Codacy said it was using the config file from our repo.  I didn't think it'd be wise for me to change that config just to placate Codacy.  At least, not without discussing it with you and the team first.  That might mask real errors, I'd guess.